### PR TITLE
Include module with type name in validation errors (python)

### PR DESCRIPTION
**Pull request by connorworley**
This is helpful for debugging when types that share the same name but live in different namespaces are mistakenly used with each other.

### CODE REVIEW

File: stone/backends/python_rsrc/stone_validators.py [hunk 1]
```diff
@@ -69,6 +69,11 @@ def __repr__(self):
         return 'ValidationError(%r)' % six.text_type(self)
 
 
+def type_name_with_module(t):
+    # type: (typing.Type[typing.Any]) -> six.text_type
+    return '%s.%s' % (t.__module__, t.__name__)
+
+
 def generic_type_name(v):
     """Return a descriptive type name that isn't Python specific. For example,
```
Comment (KarandeepJohar) [inline stone/backends/python_rsrc/stone_validators.py:78]: can you add types here as well?

File: stone/backends/python_rsrc/stone_validators.py [hunk 2]
```diff
@@ -69,6 +69,11 @@ def __repr__(self):
         return 'ValidationError(%r)' % six.text_type(self)
 
 
+def type_name_with_module(t):
```
Comment (KarandeepJohar) [inline stone/backends/python_rsrc/stone_validators.py:72]: shouldn't we always use `generic_type_name` instead of this right?

**Review by KarandeepJohar — CHANGES_REQUESTED** [review]

File: stone/backends/python_rsrc/stone_validators.py (hunk 2 above)
Comment (connorworley) [reply stone/backends/python_rsrc/stone_validators.py:72]: I don't think that `generic_type_name` will work for the `self.definition` case later on - it'll end up returning `builtins.type`.

**Review by rogebrd — APPROVED** [review]
LGTM

**Pull request merged**